lib/intern.ts

```diff
@@ -0,0 +1,211 @@
+export class intern {
+  static is_new(ent: any): boolean {
+    // NOTE: This function is intended for use by the #save method. This
+    // function returns true when the entity argument is assumed to not yet
+    // exist in the store.
+    //
+    // In terms of code, if client code looks like so:
+    // ```
+    //   seneca.make('product')
+    //     .data$({ label, price })
+    //     .save$(done)
+    // ```
+    //
+    // - `is_new` will be invoked from the #save method and return
+    // true, because the product entity is yet to be saved.
+    //
+    // The following client code will cause `is_new` to return false,
+    // when invoked from the #save method, because the user entity already
+    // exists:
+    // ```
+    //   seneca.make('user')
+    //     .load$(user_id, (err, user) => {
+    //       if (err) return done(err)
+    //
+    //       return user
+    //         .data$({ email, username })
+    //         .save$(done)
+    //     })
+    // ```
+    // 
+    return null != ent && null == ent.id
+  }
+
+
+  static is_upsert_requested(msg: any): boolean {
```

Comment:
https://github.com/senecajs/seneca-mem-store/pull/88#discussion_r636051265
>  lilsweetcaligula 5 days ago Author Collaborator
However, the `_requested` part indicates that the argument is not an entity, as could otherwise be assumed. Naming the utility `is_upsert` would imply that the first argument is an entity.